File: src/transformers/features/misc/MiscIsPruneTransformer.ts
import { MiscIsPruneProgrammer } from "../../../programmers/misc/MiscIsPruneProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscIsPruneTransformer {
  export const transform = GenericTransformer.scalar("misc.isPrune")(
    (project) => (modulo) => MiscIsPruneProgrammer.write(project)(modulo),
  );
}

File: packages/transform/src/features/misc/MiscIsPruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscIsPruneTransformer = miscIsPruneTransformerNamespace{}

type miscIsPruneTransformerNamespace struct{}

func (miscIsPruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	prune := coremisc.MiscPruneProgrammer.Write(call.TypeText)
	return "((input) => { " + prune + "(input); return true; })"
}
